test/isolaterunner_test.dart

```diff
@@ -71,40 +72,47 @@ void testIsolateFunctions() {
       }).whenComplete(isolate.close);
     });
   });
-  test("pause2", () {
+
+  test("pause2", () async {
     Capability c1 = new Capability();
     Capability c2 = new Capability();
     int mayCompleteCount = 2;
-    return IsolateRunner.spawn().then((isolate) {
-      isolate.pause(c1);
-      isolate.pause(c2);
-      new Future.delayed(MS * 500, () {
-        mayCompleteCount--;
-        isolate.resume(c1);
-      });
-      new Future.delayed(MS * 500, () {
-        mayCompleteCount--;
-        isolate.resume(c2);
-      });
-      isolate.run(id, 42).then((v) {
-        expect(v, 42);
-        expect(mayCompleteCount, 0);
-      }).whenComplete(isolate.close);
+
+    var isolate = await IsolateRunner.spawn();
+
+    isolate.pause(c1);
+    isolate.pause(c2);
+
+    new Future.delayed(MS * 500, () {
+      mayCompleteCount--;
+      isolate.resume(c1);
     });
-  });
-  test("ping", () {
-    return IsolateRunner.spawn().then((isolate) {
-      return isolate.ping().then((v) {
-        expect(v, isTrue);
-        return isolate.close();
-      });
+
+    new Future.delayed(MS * 500, () {
+      mayCompleteCount--;
+      isolate.resume(c2);
     });
+
+    try {
+      var v = await isolate.run(id, 42);
+      expect(v, 42);
+      expect(mayCompleteCount, 0);
+    } finally {
+      await isolate.close();
+    }
   });
-  test("kill", () {
-    return IsolateRunner.spawn().then((isolate) {
-      return isolate.kill();
-    });
+
+  test("ping", () async {
+    var isolate = await IsolateRunner.spawn();
+    var v = await isolate.ping();
+    expect(v, isTrue);
+    await isolate.close();
   });
+
+  test("kill", () async {
+    var isolate = await IsolateRunner.spawn();
+    await isolate.kill();
+  }, tags: 'skip_Dart_v1_x');
```

Comment:
Why does this test fail?

Reply:
Zero clue. It just times out. Would love someone to take a look...